tests/test_app.py

```diff
@@ -0,0 +1,331 @@
+
+from girder_worker.app import (
+    GirderAsyncResult,
+    Task,
+    _maybe_model_repr
+)
+
+
+RESERVED_HEADERS = [
+    ('girder_client_token', 'GIRDER_CLIENT_TOKEN'),
+    ('girder_api_url', 'GIRDER_API_URL'),
+    ('girder_result_hooks', 'GIRDER_RESULT_HOOKS')
+]
+
+RESERVED_OPTIONS = [
+    ('girder_user', 'GIRDER_USER'),
+    ('girder_job_title', 'GIRDER_JOB_TITLE'),
+    ('girder_job_type', 'GIRDER_JOB_TYPE'),
+    ('girder_job_public', 'GIRDER_JOB_PUBLIC'),
+    ('girder_job_handler', 'GIRDER_JOB_HANDLER'),
+    ('girder_job_other_fields', 'GIRDER_JOB_OTHER_FIELDS')
+]
+
+
+# This function returns a Task() object and tweaks the task's
+# request_stack to include a Context object. args and kwargs are
+# passed directly to the generated Context object. See:
+# celery.app.task.Context for more info.
+def _task_with_request(*args, **kwargs):
+    task = Task()
+    task.request_stack = celery.utils.threads.LocalStack()
+    task.push_request(*args, **kwargs)
+    return task
+
+
+class MockTrans(object):
+    def __init__(self, arg):
+        self.arg = arg
+
+    def transform(self, *args, **kwargs):
+        return self.arg
+
+
+class MockNonTrans(object):
+    pass
+
+
+MockNonTransObj = MockNonTrans()
+
+
+# Note: This test checks whether the
+# girder_worker.app.Task.reserved_headers are the same as the headers
+# defined in RESERVED_HEADERS. If it is failing it is probably because
+# you added a reserved header to girder_worker.app.Task and not to
+# RESERVED_HEADERS
+def test_TASK_reserved_headers_same_as_test_reserved_headers():
+    assert set(Task.reserved_headers) == set([h for h, _ in RESERVED_HEADERS])
+
+
+# Note: This test checks whether the
+# girder_worker.app.Task.reserved_options are the same as the options
+# defined in RESERVED_OPTIONS. If it is failing it is probably because
+# you added a reserved options to girder_worker.app.Task and not to
+# RESERVED_OPTIONS
+def test_TASK_reserved_options_same_as_test_reserved_options():
+    assert set(Task.reserved_options) == set([h for h, _ in RESERVED_OPTIONS])
+
+
+def test_GirderAsyncResult_job_property_returns_None_on_ImportError():
+    gar = GirderAsyncResult('BOGUS_TASK_ID')
+    assert gar.job is None
+
+
+def test_GirderAsyncResult_job_property_calls_findOne_on_girder_job_model():
+    gar = GirderAsyncResult('BOGUS_TASK_ID')
+    with mock.patch('girder.utility.model_importer.ModelImporter') as mi:
+        gar.job
+        mi.model.return_value.findOne.assert_called_once_with({
+            'celeryTaskId': 'BOGUS_TASK_ID'
+        })
+
+
+def test_GirderAsyncResult_job_property_returns_None_if_no_jobs_found():
+    gar = GirderAsyncResult('BOGUS_TASK_ID')
+    with mock.patch('girder.utility.model_importer.ModelImporter') as mi:
+        mi.model.return_value.findOne.side_effect = IndexError()
+        assert gar.job is None
+
+
+def test_Task_AsynResult_of_type_GirderAsyncResult():
+    assert isinstance(Task().AsyncResult('BOGUS_TASK_ID'), GirderAsyncResult)
```

Comment:
Async*

Does it make sense to stub out a `pytest.mark.skip('unimplemented')` test here which is retrieving the `job` property when Girder isn't importable (on a remote worker)?